### PR TITLE
Run test suite with GitHub Actions

### PR DESCRIPTION
**Pull request by kevinmbeaulieu**

### CODE REVIEW

**Comment by kevinmbeaulieu** [comment]
Btw looks like the pod install step takes awhile (~14m), so I'll look into any way to cache that as a follow-up to this

**Comment by danoc** [comment]
> Btw looks like the pod install step takes awhile (~14m), so I'll look into any way to cache that as a follow-up to this

I've used GitHub's cache action which worked really well: https://github.com/actions/cache

**Comment by danoc** [comment]
I don't know if you're going to be adding linters and other types of checks, but one thing that's worked well for us is to break those up into separate jobs in the same workflow file. They run in parallel which, combined with the caching, leads to very fast results:

image

**Comment by kevinmbeaulieu** [comment]
Actually just updated this PR with the caching- brings the total CI time down from 15min to only 5min 🎉

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+name: CI
+on: push
+jobs:
+  test:
+    name: Run tests
+    runs-on: macos-latest
```
Comment (scottasoutherland) [inline .github/workflows/ci.yml:6]: Do we want to use a specific mac-os instead of macos-latest ? (For some reason macos-latest is 10.15 ? )

**Review by scottasoutherland — APPROVED** [review]
LGTM, just one question

File: .github/workflows/ci.yml (hunk 1 above)
Comment (kevinmbeaulieu) [reply .github/workflows/ci.yml:6]: looks like this is why macos-latest is 10.15, but i'll switch to explicitly run on 10.15 so we don't get any surprises when the default switches to 11.0 
https://github.com/actions/virtual-environments
> The macOS 11.0 virtual environment is currently provided as a private preview only.
> The "macos-latest" YAML workflow label still uses the macOS 10.15 virtual environment.

**Pull request merged**